lib/data_explorer_query_group_bookmarkable.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+class DataExplorerQueryGroupBookmarkable < BaseBookmarkable
+  def self.model
+    DataExplorer::QueryGroup
+  end
+
+  def self.serializer
+    UserDataExplorerQueryGroupBookmarkSerializer
+  end
+
+  def self.preload_associations
+    [:data_explorer_queries, :groups]
+  end
+
+  def self.list_query(user, guardian)
+    group_ids = user.groups.pluck(:id) 
+    return if group_ids.empty?
+    user.bookmarks_of_type("DataExplorer::QueryGroup")
+    .joins("INNER JOIN data_explorer_query_groups ON data_explorer_query_groups.id = bookmarks.bookmarkable_id
+          AND bookmarks.bookmarkable_type = 'DataExplorer::QueryGroup'")
```

Comment:
I don't think this line is necessary since `bookmarks_of_type` is already doing the same thing.

Reply:
I was curious about this, but forgot to put a TODO for myself to check. I based this off of ChatMessageBookmarkable, which it turns out is similarly redundant. I'll confirm and update.

Reply:
Yes makes sense, you are right :+1:

Reply:
> It seems this INNER JOIN is necessary, since bookmarks_of_type only returns the bookmark record, so there's still a need to join the bookmarkable's table.

Sorry what I meant is that this filter `bookmarks.bookmarkable_type = 'DataExplorer::QueryGroup'"`, is not necessary because "user.bookmarks_of_type" adds the same exact filter. I think the PG planner should be smart enough to only join the relevant `bookmarks` rows to the `data_explorer_query_groups`.

Reply:
Oh yeah, you're right, I missed that. Thanks!